src/database.py

```diff
@@ -0,0 +1,144 @@
+        """
+        conn = self.get_conn()
+        database_name = self.get_database_name()
+        if conn is not None:
+            try:
+                with conn.cursor() as curs:
+                    curs.execute(
+                        "UPDATE pg_database "
+                        "SET datcollate='C', datctype='C' "
+                        "WHERE datname=%s",
+                        (database_name,),
+                    )
+            except psycopg2.Error as exc:
+                logger.error("Failed to prepare database: %s", str(exc))
+                raise
+
+    def erase_database(self) -> None:
+        """Erase database.
+
+        Raises:
+           Error: something went wrong while erasing the database.
+        """
+        conn = self.get_conn("template1")
+        database_name = self.get_database_name()
+        if conn is not None:
+            try:
+                with conn.cursor() as curs:
+                    curs.execute(sql.SQL("DROP DATABASE {}").format(sql.Identifier(database_name)))
+                    curs.execute(
+                        sql.SQL(
+                            "CREATE DATABASE {} "
+                            "WITH LC_CTYPE = 'C' LC_COLLATE='C' TEMPLATE='template0';"
+                        ).format(sql.Identifier(database_name))
+                    )
+            except psycopg2.Error as exc:
+                logger.error("Failed to erase database: %s", str(exc))
+                raise
+
+    def get_conn(self, database_name: str = "") -> connection | None:
+        """Erase database.
+
+        Args:
+            database_name: database to connect to.
+
+        Raises:
+           Error: something went wrong while erasing the database.
+
+        Returns:
+            Connection or None if there is no relation data
+        """
+        relation_data = self.get_relation_data()
+        if relation_data is not None:
+            try:
+                user = relation_data.get("POSTGRES_USER")
+                password = relation_data.get("POSTGRES_PASSWORD")
+                host = relation_data.get("POSTGRES_HOST")
+                if not database_name:
+                    database_name = self.get_database_name()
+                conn = psycopg2.connect(
+                    f"dbname='{database_name}' user='{user}' host='{host}'"
+                    f"password='{password}' connect_timeout=1"
+                )
+                conn.autocommit = True
+                return conn
+            except psycopg2.Error as exc:
+                logger.error("Failed to connect to database: %s", str(exc))
+                raise
+        return None
+
+    def get_database_name(self) -> str:
+        """Get database name from relation.
+
+        Returns:
+            database name
+        """
+        relation_data = self.get_relation_data()
+        if relation_data is not None:
+            return relation_data.get("POSTGRES_DB", "")
+        return ""
```

Comment:
Would it be confusing to get an empty string if there is no database configured? Why not `None`?

Reply:
I think this is the same case that you asked before regarding the connection. Instead of returning None/empty, would be better to block the charm since a relation is being required and is not available?

Reply:
I'll mark this one as resolved so we can discuss this at the last comment, ok? To make it easier to track.